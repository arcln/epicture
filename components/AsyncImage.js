import React from 'react';
import {View, Image} from 'react-native';

export default class AsyncImage extends React.Component {

  state = {
    loaded: false,
  };

  render() {
    return (
      <View>
        <Image
          style={this.props.style}
          source={this.props.source}
          onLoad={() => this.setState({loaded: true})}
        />
        {!this.state.loaded && this.props.placeholder}
      </View>
    )
  }
}