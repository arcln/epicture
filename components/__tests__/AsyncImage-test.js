import 'react-native';
import React from 'react';
import AsyncImage from '../AsyncImage';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <AsyncImage
      source={{uri: 'https://i.imgur.com/AkRbORU.jpg'}}
      width={300}
      style={{width: 300, height: 300}}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
