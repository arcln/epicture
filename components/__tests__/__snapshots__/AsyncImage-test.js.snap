// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Image
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://i.imgur.com/AkRbORU.jpg",
      }
    }
    style={
      Object {
        "height": 300,
        "opacity": 0,
        "width": 300,
      }
    }
  />
</View>
`;
